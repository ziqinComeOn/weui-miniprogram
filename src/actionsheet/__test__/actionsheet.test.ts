import path from 'path'
import simulate from 'miniprogram-simulate'

describe('actionsheet', () => {
    let id: string

    beforeAll(() => {
        id = simulate.load(path.resolve(__dirname, './index'))
    })

    test('basic actionsheet', async () => {
        const container = simulate.render(id)
        container.attach(document.createElement('parent-wrapper'))

        expect(container.toJSON()).toMatchSnapshot()
    })

    test('no cancel', async () => {
        const container = simulate.render(id, { showCancel: false })
        container.attach(document.createElement('parent-wrapper'))

        expect(container.toJSON()).toMatchSnapshot()
    })

    test('touch', async () => {
        const container = simulate.render(id)
        container.attach(document.createElement('parent-wrapper'))
        const comp = container.querySelector('.comp')

        const menu = comp.querySelectorAll('.weui-actionsheet__menu .weui-actionsheet__cell')

        menu[2].dispatchEvent('touchstart')
        menu[2].dispatchEvent('touchend')
        await simulate.sleep(10)
        expect(container.data.tapValue).toBe(3)

        menu[1].dispatchEvent('touchstart')
        menu[1].dispatchEvent('touchend')
        await simulate.sleep(10)
        expect(container.data.tapValue).toBe(2)
    })

    test('close', async () => {
        const container = simulate.render(id)
        container.attach(document.createElement('parent-wrapper'))
        const comp = container.querySelector('.comp')

        const cancel = comp.querySelector('.weui-actionsheet__action .weui-actionsheet__cell')

        cancel.dispatchEvent('touchstart')
        cancel.dispatchEvent('touchend')
        await simulate.sleep(10)
        expect(container.data.show).toBe(false)
        expect(comp.data.show).toBe(false)
    })
})
