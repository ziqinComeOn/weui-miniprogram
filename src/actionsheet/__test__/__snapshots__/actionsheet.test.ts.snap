// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`actionsheet basic actionsheet 1`] = `
<main>
  <mp-actionsheet
    class="comp"
    bind:actiontap="actiontap"
    bind:close="close"
  >
    <wx-view
      class="weui-mask  customer-mask-class"
      bind:tap="closeActionSheet"
    />
    <wx-view
      class="weui-actionsheet weui-actionsheet_toggle "
    >
      <wx-view
        class="weui-actionsheet__title"
      >
        <wx-view
          class="weui-actionsheet__title-text"
        >
          actionsheet title
        </wx-view>
      </wx-view>
      <wx-view
        class="weui-actionsheet__menu"
      >
        <wx-view
          class="weui-actionsheet__cell "
          data-groupindex="{{0}}"
          data-index="{{0}}"
          data-value="{{1}}"
          bind:tap="buttonTap"
        >
          
                item 1
            
        </wx-view>
        <wx-view
          class="weui-actionsheet__cell "
          data-groupindex="{{0}}"
          data-index="{{1}}"
          data-value="{{2}}"
          bind:tap="buttonTap"
        >
          
                item 2
            
        </wx-view>
        <wx-view
          class="weui-actionsheet__cell weui-actionsheet__cell_warn"
          data-groupindex="{{0}}"
          data-index="{{2}}"
          data-value="{{3}}"
          bind:tap="buttonTap"
        >
          
                item 3
            
        </wx-view>
      </wx-view>
      <wx-view
        class="weui-actionsheet__action"
      >
        <wx-view
          class="weui-actionsheet__cell"
          data-type="close"
          id="iosActionsheetCancel"
          bind:tap="closeActionSheet"
        >
          actionsheet cancel text
        </wx-view>
      </wx-view>
    </wx-view>
  </mp-actionsheet>
</main>
`;

exports[`actionsheet no cancel 1`] = `
<main>
  <mp-actionsheet
    class="comp"
    bind:actiontap="actiontap"
    bind:close="close"
  >
    <wx-view
      class="weui-mask  customer-mask-class"
      bind:tap="closeActionSheet"
    />
    <wx-view
      class="weui-actionsheet weui-actionsheet_toggle "
    >
      <wx-view
        class="weui-actionsheet__title"
      >
        <wx-view
          class="weui-actionsheet__title-text"
        >
          actionsheet title
        </wx-view>
      </wx-view>
      <wx-view
        class="weui-actionsheet__action"
      >
        <wx-view
          class="weui-actionsheet__cell "
          data-groupindex="{{0}}"
          data-index="{{0}}"
          data-value="{{1}}"
          bind:tap="buttonTap"
        >
          
                item 1
            
        </wx-view>
        <wx-view
          class="weui-actionsheet__cell "
          data-groupindex="{{0}}"
          data-index="{{1}}"
          data-value="{{2}}"
          bind:tap="buttonTap"
        >
          
                item 2
            
        </wx-view>
        <wx-view
          class="weui-actionsheet__cell weui-actionsheet__cell_warn"
          data-groupindex="{{0}}"
          data-index="{{2}}"
          data-value="{{3}}"
          bind:tap="buttonTap"
        >
          
                item 3
            
        </wx-view>
      </wx-view>
    </wx-view>
  </mp-actionsheet>
</main>
`;
